src/dstack/_internal/server/background/tasks/process_submitted_jobs.py

```diff
@@ -74,21 +86,111 @@ async def _process_submitted_job(session: AsyncSession, job_model: JobModel):
     )
     run_model = res.scalar_one()
     project_model = run_model.project
+
+    # check default pool
+    pool = project_model.default_pool
+    if pool is None:
+        # TODO: get_or_create_default_pool...
+        pools = await list_project_pool_models(session, job_model.project)
+        for pool_item in pools:
+            if pool_item.id == job_model.project.default_pool_id:
+                pool = pool_item
+            if pool_item.name == DEFAULT_POOL_NAME:
+                pool = pool_item
+        if pool is None:
+            pool = PoolModel(
```

Comment:
Refactor pool creation. You already have `create_pool_model()`.

Reply:
Fixed